crates/control/src/localization.rs

```diff
@@ -16,48 +16,18 @@ use types::{
     localization::{ScoredPose, Update},
     multivariate_normal_distribution::MultivariateNormalDistribution,
     CorrespondencePoints, Direction, FieldDimensions, FieldMark, GameControllerState, InitialPose,
-    Line, Line2, LineData, Players, PrimaryState, Side,
+    Line, Line2, LineData, Players, PrimaryState, PrimaryStateTransition, Side,
 };
 
 pub struct Localization {
     field_marks: Vec<FieldMark>,
-    last_primary_state: PrimaryState,
     hypotheses: Vec<ScoredPose>,
-    hypotheses_when_entered_playing: Vec<ScoredPose>,
-    is_penalized_with_motion_in_set: bool,
-    was_picked_up_while_penalized_with_motion_in_set: bool,
+    last_penalty: Option<Penalty>,
```

Comment:
Is `last` the right term here? Is it from the previous cycle? Why do you need to store the penalty from the previous cycle?

Reply:
Because I need the kind of penalty i had when i am entering `Playing`. But as soon as I entered playing, there is no penalty anymore